ocaml/auth/pam.ml

```diff
@@ -15,3 +15,6 @@
 external authenticate : string -> string -> unit = "stub_XA_mh_authorize"
 
 external change_password : string -> string -> unit = "stub_XA_mh_chpasswd"
+
+external crypt_r : key:string -> salt:string -> string option
```

Comment:
Might be useful to add a small comment here showing what salt should be (or call this 'setting' as the C API does). We have to be careful to include `$6$` at the beginning, otherwise we'll get a very weak hash.